### PR TITLE
PUT /session to return 401 Unauthorized error when user cannot be found

### LINKED ISSUE #52: PUT /session to return 401 Unauthorized error when user cannot be found
Currently the server returns `404 Not Found` which is not compliant with the specification:
http://docs.accountjsonapi.apiary.io/#reference/current-user/session/sign-in

---

- [x] claim this issue (comment below or assign yourself)
- [x] setup and run tests as described here: https://github.com/hoodiehq/hoodie-server-account/blob/master/tests/README.md – there should be no failing test
- [x] [uncomment failing test](https://github.com/hoodiehq/hoodie-server-account/blob/e37cbdd5d364d1d1cffc583929fd17f8ca07eedf/tests/integration/routes/session-test.js#L143-L156). Test should fail now. 
- [x] Commit with `test: PUT /session to return 401 Unauthorized error when user cannot be found` and start a Pull Request
- [x] fix the issue (commit with `fix: PUT /session to return 401 Unauthorized error when user cannot be found`)
- [x] review & merge
- [x] update https://github.com/gr2m/milestones/issues/88
- [x] [uncomment integration test](https://github.com/hoodiehq/hoodie-app-something-tracker/blob/f961de1ba467ad67d6e24cd43dad1621ee855d78/test/account-test.js#L73-L75)

If you have any questions, comment or ping us in the [Hoodie chat](http://hood.ie/chat) or on [Twitter](https://twitter.com/hoodiehq)

### PR DESCRIPTION
**Pull request by patriciagarcia**
Related #52

### CODE REVIEW

File: routes/utils/errors.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ module.exports.INVALID_SESSION = hoodieError({
   status: 401
 })
 
+module.exports.USER_NOT_FOUND = hoodieError({
+  name: 'Unauthorized',
+  message: 'Invalid credentials',
```
Comment (patriciagarcia) [inline routes/utils/errors.js]: According to the [spec](http://docs.accountjsonapi.apiary.io/#reference/current-user/session/sign-in) the error message should be "Invalid credentials", let me know if you prefer to return something else instead

**Comment by patriciagarcia** [comment]
Related: in `GET /session`, according to the [spec](http://docs.accountjsonapi.apiary.io/#reference/current-user/session/check-session) when the user is not found or the session is invalid the error returned should be 401 instead of 404, as in [here](https://github.com/hoodiehq/hoodie-server-account/blob/master/tests/integration/routes/session-test.js#L313) and [here](https://github.com/hoodiehq/hoodie-server-account/blob/master/tests/integration/routes/session-test.js#L357).

Should I create an issue with that?

**Comment by gr2m** [comment]
I feel like the spec is wrong, if I try to `GET /session` but there is no session, isn’t 404 the right error code? `GET /session` cannot return an invalid session, a session is either there or not, so I feel like 401 is the right code, feels like the way other systems do it, too? Shall we adapt the spec?

File: routes/utils/errors.js (hunk 1 above)
Comment (gr2m) [inline routes/utils/errors.js]: should we call the error `INVALID_CREDENTIALS` instead of  `USER_NOT_FOUND`?

**Comment by gr2m** [comment]
as discussed with Pat on a call, we will fix the spec for `GET /session` to return 404 instead of 401

**Comment by patriciagarcia** [comment]
As re-discussed with Gregor and after Twitter consultation https://twitter.com/gr2m/status/697804572623007744, we will keep `401` as error for `GET /session` when the session is not there.

(WIP: Create issues to fix the implementation)

**Comment by gr2m** [comment]
released in https://github.com/hoodiehq/hoodie-server-account/releases/tag/v2.6.1

**Pull request merged**